src/main/scala/workflow/DefaultOptimizer.scala

```diff
@@ -5,8 +5,9 @@ package workflow
  */
 object DefaultOptimizer extends Optimizer {
   protected val batches: Seq[Batch] =
-    Batch("DAG Optimization", FixedPoint(100), EquivalentNodeMergeRule) ::
+    Batch("DAG Optimization", FixedPoint(Int.MaxValue), EquivalentNodeMergeRule) ::
```

Comment:
Given it's a DAG, assuming the equivalent node merging isn't altering the plan when there's nothing to merge it should be O(max depth of DAG). I do need to check the node merging to make sure though.